tests/testsuite/edition.rs

```diff
@@ -122,3 +122,57 @@ fn edition_unstable() {
         )
         .run();
 }
+
+#[cargo_test]
+fn unset_edition_works_with_no_newer_compatible_edition() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = 'foo'
+                version = '0.1.0'
+                rust-version = "1.0"
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("check -v")
+        .with_stderr(
+            "\
+[WARNING] no edition set: defaulting to the 2015 edition
+[CHECKING] foo [..]
+[RUNNING] `rustc [..] --edition=2015 [..]`
+[FINISHED] [..]
+",
+        )
+        .run();
+}
+
+#[cargo_test]
+fn unset_edition_works_on_old_msrv() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = 'foo'
+                version = '0.1.0'
+                rust-version = "1.50"  # contains 2018 edition
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("check -v")
+        .with_stderr(
+            "\
+[WARNING] no edition set: defaulting to the 2015 edition while 2018 is compatible with `rust-version`
```

Comment:
Should we call out which package it is? There might be a workspace containing tons of members.

Reply:
Looks like its done for us:
```
                        // In a workspace, it can be confusing where a warning
                        // originated, so include the path.
                        format!("{}: {}", path.display(), warning.message)
```

The other warnings I looked at do not mention the package which is why I dug into it to find that code.  The test case must be only for a single package so it gets elided.